Modules/Core/Common/src/otbStopwatch.cxx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2005-2017 Centre National d'Etudes Spatiales (CNES)
+ *
+ * This file is part of Orfeo Toolbox
+ *
+ *     https://www.orfeo-toolbox.org/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "otbStopwatch.h"
+
+namespace otb
+{
+
+Stopwatch
+::Stopwatch()
+    : m_ElapsedMilliseconds(), m_IsRunning()
+{
+}
+
+void
+Stopwatch
+::Start()
+{
+  if (!this->m_IsRunning)
+    {
+    this->m_IsRunning = true;
+    this->m_StartTime = std::chrono::steady_clock::now();
+    }
+}
+
+void
+Stopwatch
+::Stop()
+{
+  if (this->m_IsRunning)
+    {
+    this->m_ElapsedMilliseconds += GetRunningElapsedTime();
+    this->m_IsRunning = false;
+    }
+}
+
+void
+Stopwatch
+::Reset()
+{
+  this->m_ElapsedMilliseconds = 0;
+  this->m_IsRunning = false;
+}
+
+void
+Stopwatch
+::Restart()
+{
+  this->m_ElapsedMilliseconds = 0;
+  this->m_IsRunning = true;
+  this->m_StartTime = std::chrono::steady_clock::now();
+}
+
+Stopwatch::ElapsedTimeType
+Stopwatch
+::GetElapsedMilliseconds() const
+{
+  auto result = this->m_ElapsedMilliseconds;
+
+  if (this->m_IsRunning)
+    result += this->GetRunningElapsedTime();
+
+  return result;
+}
+
+bool
+Stopwatch
+::GetIsRunning() const
+{
+  return this->m_IsRunning;
+}
+
+Stopwatch
+Stopwatch
+::StartNew()
+{
+  Stopwatch sw;
+  sw.Start();
+
+  return sw;
+}
+
+Stopwatch::ElapsedTimeType
```

Comment:
Since you've moved toward C++11 chrono, why don't you return a duration from here?

Reply:
Do you mean specifically in that function (it's private and only called twice), or in the API? The private function could probably return `std::chrono::milliseconds` without any issues.

I didn't want to bring in `std::chrono::duration` into the API to keep things simple for the callers, but I don't feel strongly either way.